zerver/actions/message_edit.py

```diff
@@ -194,10 +197,10 @@ def send_message_moved_breadcrumbs(
     user_profile: UserProfile,
     old_stream: Stream,
     old_topic: str,
-    old_thread_notification_string: Optional[str],
+    old_thread_notification_string: Union[str, "StrPromise", None],
     new_stream: Stream,
     new_topic: Optional[str],
-    new_thread_notification_string: Optional[str],
+    new_thread_notification_string: Union[str, "StrPromise", None],
```

Comment:
These are always `Optional[StrPromise]` and cannot be `str`. This is important because they’re used inside `override_language` blocks.

Reply:
Fixed.